### PR TITLE
Audio Datasets

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
Blog post to promote audio datasets in 🤗 Datasets and on the HF Hub.

<details>
<summary> Proposed structure: </summary>

**Intro**
Explain motivations: plant the seed that 🤗 Datasets is the number one place for downloading and preparing audio datasets.

**Sec 1: Loading a Dataset**

- Show how easy it is to download an audio dataset in one line of Python code
- Explain briefly the args to the `load_dataset` function
- Explain what's going on under the hood: download, extract, prepare data & splits

**Sec 2: Easy to Load, Easy to Process**

- Example for preparing an audio dataset for inference / fine-tuning
- Use `AutoProcessor` to load a processor from any checkpoint
- How to resample to a desired sampling rate
- How to pre-process the audio inputs (feature extractor) and tokenise the transcriptions

**Sec 3: The Hub**

- Datasets Card -> dataset info, everything you need to know about the dataset in one place
- Highlight Datasets Viewer -> the ease with which datasets can be viewed/experienced
- Repos for storing raw audio data

**Sec 4: A Tour of Audio Datasets on the Hub**
Synopsis of the major audio datasets.

- English ASR (ESB datasets)
- Multilingual ASR
- Speech Translation
- Audio Classification

+streaming mode -> think about which section this falls best under or an entirely new section

</details>

### CODE REVIEW

**Comment by Vaibhavs10** [comment]
A couple of suggestions:
1. Loading a Dataset: We should also talk about Audio Folder here. Leveraging local datasets would be a key part for researchers, hobbyists, and professionals alike.
2. Easy to load: Maybe we extend this a bit and talk about streaming a bit here. This way we nail the point that datasets are quite resilient and can do whatever they want.
3. Process: We should definitely talk about map and cast_column here, this will be quite useful for snipping audio/ adding noise/ etc. One function to rule them all.

WDYT?

**Comment by sanchit-gandhi** [comment]
1. We could feature the audio folder prominently on the Google Colab example too -> would be well suited here in the 'hands on' accompaniment
2. Yep, streaming will slot into the loading section most probably
3. Process: absolutely, will highlight `cast_column` and `map` through pre-processing examples. I'm not sure if data aug is the way to go though - I had in mind the more conventional steps of resampling (`cast_column`) and preparing a dataset for a model (`map`)

File: audio-datasets.md [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+ 'url': 'https://www.youtube.com/watch?v=zr2n1fLVasU', 
+ 'source': 2, 
+ 'category': 24, 
+ 'original_full_path': 'audio/youtube/P0004/YOU0000000315.opus',
+ }
+```
+
+We can see that there are a number of features returned by the `DatasetDict`, including `segment_id`, `speaker`, `text`, 
+`audio` and more. For speech recognition, we'll be concerned with the `text` and `audio` columns.
+
```
Comment (sanchit-gandhi) [inline audio-datasets.md:149]: Potentially missing a (sub-) section on Audio Folder - could be nice to showcase how one can upload their own audio data to the Hub

**Comment by sanchit-gandhi** [comment]
Colab link: https://colab.research.google.com/drive/1ySCQREPZEl4msLfxb79pYYOWjUZhkr9y
Will move to GH once finalised.

File: audio-datasets.md [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+        <div class="bfc">
+            <code>sanchit-gandhi</code>
+            <span class="fullname">Sanchit Gandhi</span>
+        </div>
+    </a>
+</div>
+
+<!--- Note to reviewer: comments and TODOs are included in this format. --->
+
+<!--- TODO: insert link to Colab --->
```
Comment (sanchit-gandhi) [inline audio-datasets.md:27]: Colab link: https://colab.research.google.com/drive/1ySCQREPZEl4msLfxb79pYYOWjUZhkr9y
Will move to GH once finalised.

**Review by osanseviero — COMMENTED** [review]
Very very cool blog post 🔥

File: audio-datasets.md [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+
+How can you enable streaming mode? Easy! Just set `streaming=True` when you load your dataset. The rest will be taken 
+care for you:
+
+```python
+gigaspeech = load_dataset("speechcolab/gigaspeech", "xs", streaming=True)
+```
+
+All the steps covered so far in this tutorial can be applied to the streaming dataset without any code changes.
+<!--- TODO: make this statement valid --->
```
Comment (lhoestq) [inline audio-datasets.md:347]:
```suggestion
The only change is that you can no longer access one sample in particular using `dataset[sample_idx]`, but instead have to iterate over the dataset (using a for loop for example).
```

File: audio-datasets.md [hunk 4]
```diff
@@ -0,0 +1,569 @@
...
+
+#### [Common Voice](https://huggingface.co/datasets/mozilla-foundation/common_voice_11_0)
+Common Voice is a series of crowd-sourced open-licensed speech datasets where speakers record text from Wikipedia in 
+various languages. Since anyone can contribute recordings, there is significant variation in both audio quality and 
+speakers. The audio conditions are challenging, with recording artefacts, accented speech, hesitations, and the presence 
+of foreign words. The transcriptions are both cased and punctuated. The English subset of version 11.0 contains 
+approximately 2,300 hours of validated data.
+
+```python
+common_voice = load_dataset("mozilla-foundation/common_voice_11", "en", use_auth_token=True)
```
Comment (lhoestq) [inline audio-datasets.md:437]: maybe mention why you need a token to access this one

**Review by lhoestq — COMMENTED** [review]
Awesome thank you !

File: audio-datasets.md (hunk 2 above)
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/5c797aa54cb6196c06d7389a86842e76255f7ba8

File: audio-datasets.md [hunk 5]
```diff
@@ -0,0 +1,569 @@
...
+
+🤗 Datasets is an open-source library for downloading and preparing datasets of all domains. Its minimalistic API 
+allows users to download and prepare datasets in just one line of Python code, with a suite of functions that 
+enable efficient pre-processing. The number of datasets available is unparalleled, with all the most popular 
+audio datasets available to download. 
+
+Not only this, but 🤗 Datasets comes prepared with multiple audio-specific features that make working 
+with audio datasets easy for researchers and practitioners alike. In this blog, we'll demonstrate how 🤗 Datasets 
+is the number one place for downloading and preparing audio datasets. Carry on reading to find out how to load and 
+prepare the most popular audio datasets in just one line of Python code.
```
[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply audio-datasets.md:41]: We'll that's a bit the message we want to convey to people though IMO :sweat_smile:

[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply audio-datasets.md:41]: But yeah maybe a bit aggressive

File: audio-datasets.md [hunk 6]
```diff
@@ -0,0 +1,565 @@
...
+
+## Contents
+1. [Load an Audio Dataset](#load-an-audio-dataset)
```
Comment (patrickvonplaten) [inline audio-datasets.md:44]: actually before jumping into code, I think we should present the Hub first, that's more visually and a "softer" introduction I think. 

E.g. I'd start with the "Hub" section, then brag about all the datasets we have with the nice list you made, then maybe add one sentence to explain how people can find other audio datasets using the task categories.

And only then jump into the code.

E.g. people first need to find the dataset they want to load and this is a very important step!

**Review by patrickvonplaten — COMMENTED** [review]
Looks very clean, but I would change the order a bit. I think "The Hub" and "Tour of audio datasets" should go in the beginning. That's the most exciting part, it's always "softer" to first see visuals and demos and then the code.

File: audio-datasets.md [hunk 7]
```diff
@@ -0,0 +1,565 @@
...
+
+One of the biggest challenges faced with audio datasets is their sheer size. The GigaSpeech `xs` split contained just 10 
+hours of training data, but amassed over 13GB of storage space for download and preparation. So what happens when we 
+want to train on a larger split? The full `xl` split contains 10,000 hours of training data, requiring over 1TB of 
+storage space! For most speech researchers, this well exceeds the specifications of a typical hard drive disk. 
+Do we need to fork out and buy additional storage? Or is there a way we can train on these datasets with **less than 
+10GB** of disk space?
+
+🤗 Datasets allows us to do just this. It is made possible through use of [_streaming_](https://huggingface.co/docs/datasets/stream), 
+depicted graphically in Figure 1. Streaming allows us to download data progressively as we iterate over the dataset:
```
Comment (polinaeterna) [inline audio-datasets.md:321]: strictly speaking, data is not being downloaded in streaming mode, as it's not saved on disk. it's just opened and iterated over

File: audio-datasets.md [hunk 8]
```diff
@@ -0,0 +1,565 @@
...
+next one.
+
+<figure>
+<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/streaming.gif" alt="Trulli" style="width:100%">
+<figcaption align = "center"><b>Figure 1:</b> Streaming mode. The dataset is divided into smaller subsets (or chunks), with subsets downloaded progressively as we iterate over the dataset. Subsets are downloaded when required and deleted after use.</figcaption>
+</figure>
+
+Streaming mode has three primary advantages over downloading the entire dataset at once:
+1. **Disk space:** you have complete control over the size of the chunks downloaded, so you can download chunks of data that will fit your device.
+2. **Download and processing speed:** audio datasets are large and need a significant amount of time to download. With streaming, downloading is done on the fly, meaning you can start using the dataset as soon as the first chunk is ready.
```
Comment (polinaeterna) [inline audio-datasets.md:334]:
```suggestion
2. **Processing time:** audio datasets are large and need a significant amount of time to download. With streaming, you don't need to spend time on data download, meaning you can start iterating over examples straightaway.

File: audio-datasets.md [hunk 9]
```diff
@@ -0,0 +1,565 @@
...
+3. **Easy experimentation:** you can experiment on one chunk of data to check that your script works without having to download the entire dataset.
+
```
Comment (polinaeterna) [inline audio-datasets.md:336]: maybe worth noting that benefits of streaming result in its drawback - as the data is not downloaded and cached and prepared in arrow on disk, reusing the dataset would require yielding from streamed file(s) again and eventually is slower than loading and generating for arrow tables. 

but maybe we can skip this :D

**Review by polinaeterna — CHANGES_REQUESTED** [review]
Love it! 

I've left a few mostly technical comments. My main suggestion is to reformulate streaming explanation a bit so that it sounds correct but without diving into technical details. Basically, we don't have any chunks of data that's being iteratively downloaded - we just stream example one by one progressively not downloading anything locally. I've left a few suggestions in some places (but probably not everywhere), feel free to reformulate them :)

File: _blog.yml [hunk 10]
```diff
@@ -1415,3 +1415,12 @@
     - nlp
     - text generation
     - research
+
+- local: audio-datasets
+  title: "A Complete Guide to Audio Datasets"
+  author: sanchit-gandhi
+  thumbnail: /blog/assets/116_audio_datasets/thumbnail.jpg
+  date: Nov 21, 2022
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply _blog.yml]: Resolved in https://github.com/huggingface/blog/pull/636/commits/1ade028af76b5c4137b3e866f192da498450eec9

File: audio-datasets.md [hunk 11]
```diff
@@ -0,0 +1,569 @@
+---
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:1]: Resolved in https://github.com/huggingface/blog/pull/636/commits/d97c1a07bcd3d3627a5769f6a33bd5066d7253be

File: audio-datasets.md [hunk 12]
```diff
@@ -0,0 +1,569 @@
...
+title: "A Complete Guide to Audio Datasets" 
+thumbnail: /blog/assets/116_audio_datasets/thumbnail.jpg
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:3]: Resolved in https://github.com/huggingface/blog/pull/636/commits/609fff09022c526a6dee83914a5e5acebaeb19ca

File: audio-datasets.md [hunk 13]
```diff
@@ -0,0 +1,569 @@
...
+---
+
+# A Complete Guide to Audio Datasets
+
+<div class="blog-metadata">
+    <small>Published 21 November, 2022.</small>
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/5798551150e14e537cf6695a8ee0473a5ec724d5

File: audio-datasets.md (hunk 6 above)
Comment (sanchit-gandhi) [reply audio-datasets.md]: Good suggestion! Moving "The Hub" section to the start. Keeping the "tour of audio datasets on the Hub" section at the end as a reference guide (people will stop reading if this is too early on).

Resolved in https://github.com/huggingface/blog/pull/636/commits/e2ec7744e3f56b28a6eea53aa24725f7a74f36a6

File: audio-datasets.md [hunk 14]
```diff
@@ -0,0 +1,565 @@
...
+excitingly, listen to actual audio samples. The Dataset Preview is loaded up in the middle of the dataset card. 
+It shows us the first 100 samples for each subset and split. What's more, it's loaded up the audio samples ready for us 
+to listen to in real-time. If we hit the play button on the first sample, we can listen to the audio and see the 
+corresponding text.
+
+The Dataset Preview is a brilliant way of experiencing audio datasets before committing to using them. 
+You can scroll through the samples for the different subsets and splits to get a better feel for the 
+audio data and gauge whether it's the right dataset for your needs.
+
+## A Tour of Audio Datasets on The Hub
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:400]: See https://github.com/huggingface/blog/pull/636#discussion_r1038755739

File: audio-datasets.md (hunk 7 above)
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/06e2200b7e77b6736d3bab7e4c43895c736b613a

File: audio-datasets.md [hunk 15]
```diff
@@ -0,0 +1,565 @@
...
+want to train on a larger split? The full `xl` split contains 10,000 hours of training data, requiring over 1TB of 
+storage space! For most speech researchers, this well exceeds the specifications of a typical hard drive disk. 
+Do we need to fork out and buy additional storage? Or is there a way we can train on these datasets with **less than 
+10GB** of disk space?
+
+🤗 Datasets allows us to do just this. It is made possible through use of [_streaming_](https://huggingface.co/docs/datasets/stream), 
+depicted graphically in Figure 1. Streaming allows us to download data progressively as we iterate over the dataset: 
+rather than downloading the whole dataset at once, we download small chunks of the dataset at a time. These chunks of 
+data are downloaded and prepared _on the fly_, meaning they are available as and when we need them, and not when we don't! 
+Once we're ready to move on to the next chunk of data, we delete the one that we currently have and move on to the
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/06e2200b7e77b6736d3bab7e4c43895c736b613a

File: audio-datasets.md [hunk 16]
```diff
@@ -0,0 +1,565 @@
...
+Once we're ready to move on to the next chunk of data, we delete the one that we currently have and move on to the 
+next one.
+
+<figure>
+<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/streaming.gif" alt="Trulli" style="width:100%">
+<figcaption align = "center"><b>Figure 1:</b> Streaming mode. The dataset is divided into smaller subsets (or chunks), with subsets downloaded progressively as we iterate over the dataset. Subsets are downloaded when required and deleted after use.</figcaption>
+</figure>
+
+Streaming mode has three primary advantages over downloading the entire dataset at once:
+1. **Disk space:** you have complete control over the size of the chunks downloaded, so you can download chunks of data that will fit your device.
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/cd6dc002d5421a235a1f6b643bc4e6f40668ad8a

File: audio-datasets.md [hunk 17]
```diff
@@ -0,0 +1,565 @@
...
+One of the biggest challenges faced with audio datasets is their sheer size. The GigaSpeech `xs` split contained just 10 
+hours of training data, but amassed over 13GB of storage space for download and preparation. So what happens when we 
+want to train on a larger split? The full `xl` split contains 10,000 hours of training data, requiring over 1TB of 
+storage space! For most speech researchers, this well exceeds the specifications of a typical hard drive disk. 
+Do we need to fork out and buy additional storage? Or is there a way we can train on these datasets with **less than 
+10GB** of disk space?
+
+🤗 Datasets allows us to do just this. It is made possible through use of [_streaming_](https://huggingface.co/docs/datasets/stream), 
+depicted graphically in Figure 1. Streaming allows us to download data progressively as we iterate over the dataset: 
+rather than downloading the whole dataset at once, we download small chunks of the dataset at a time. These chunks of
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/06e2200b7e77b6736d3bab7e4c43895c736b613a

File: audio-datasets.md (hunk 8 above)
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/d8f3f5edb7201375f60c7359803acb4408181f39

File: audio-datasets.md (hunk 9 above)
Comment (sanchit-gandhi) [reply audio-datasets.md:388]: This is a very good point! I think we should include this

Resolved in https://github.com/huggingface/blog/pull/636/commits/d8f3f5edb7201375f60c7359803acb4408181f39

File: audio-datasets.md [hunk 18]
```diff
@@ -0,0 +1,565 @@
...
+ 'audio_id': 'YOU0000000315', 
+ 'title': 'Return to Vasselheim | Critical Role: VOX MACHINA | Episode 43', 
+ 'url': 'https://www.youtube.com/watch?v=zr2n1fLVasU', 
+ 'source': 2, 
+ 'category': 24, 
+ 'original_full_path': 'audio/youtube/P0004/YOU0000000315.opus',
+ }
+```
+
+We can see that there are a number of features returned by the `DatasetDict`, including `segment_id`, `speaker`, `text`,
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/8a3f54a340ebe4cdea97a30cc9b4a254c05e3de3

File: audio-datasets.md [hunk 19]
```diff
@@ -0,0 +1,569 @@
...
+
+### 1. Resampling the Audio Data
+
+The `load_dataset` function prepares audio samples with the sampling rate that they were published with. This is not 
+always the sampling rate expected by our model. In this case, we need to _resample_ the audio to the correct sampling 
+rate.
+
+First, let's remove the dataset features not associated with speech recognition:
+
+```python
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:184]: Resolved in https://github.com/huggingface/blog/pull/636/commits/1aeea0571f2ca98de09c1ef3cbe0fcfcd732f123

File: audio-datasets.md [hunk 20]
```diff
@@ -0,0 +1,565 @@
...
+COLUMNS_TO_KEEP = ["text", "audio"]
+all_columns = next(iter(gigaspeech.values())).column_names
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Does look easier, let's do that!

I think we can skip the assertion for the purpose of the blog - 99.9% of the time datasets will have consistent features across their splits so should be ok for the most part.

File: audio-datasets.md [hunk 21]
```diff
@@ -0,0 +1,565 @@
...
+dataset, a machine translation corpus with 3001 sentence translations from English to 101 other languages. Native 
+speakers are recorded narrating the sentence transcriptions in their native languages. An \\(n\\)-way parallel corpus of 
+speech translation data is constructed by pairing the recorded audio data with the sentence transcriptions for each of the 
+101 languages. The training sets contain approximately 10 hours of supervised audio-transcription data per source-target 
+language combination.
+
+### Audio Classification
+
+#### [VoxCeleb1]()
+<!--- TODO: is this on the Hub? --->
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:561]: Is FLEURS the best one that we have then? Any other cool audio classification datasets you think are worth adding?

File: audio-datasets.md [hunk 22]
```diff
@@ -0,0 +1,565 @@
...
+speakers are recorded narrating the sentence transcriptions in their native languages. The recorded audio data is paired 
+with a label for the language in which it is spoken. The dataset can be used as an audio classification dataset for 
+_language identification_: systems are trained to predict the language of each utterance in the corpus.
+
+## Closing Remarks
+
+In this blog post, we covered how to load any audio dataset on the Hugging Face Hub with one line of Python code. We 
+performed a series of generic pre-processing steps to prepare our dataset for a machine learning model. In total, this 
+required just 13 lines of code, relying on simple Python function definitions to perform the necessary operations. We 
+introduced streaming mode, a method for downloading and preparing chunks of audio data on the fly.
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Resolved in https://github.com/huggingface/blog/pull/636/commits/511d78e0f3e559f9825a3e2cbc4ffd7e14beb20e

File: audio-datasets.md [hunk 23]
```diff
@@ -0,0 +1,565 @@
...
+| [TED-LIUM](https://huggingface.co/datasets/LIUM/tedlium)                                | TED talks                   | Oratory               | 450         | ❌      | ❌           | CC-BY-NC-ND 3.0 | Technical topics                 |
+| [GigaSpeech](https://huggingface.co/datasets/speechcolab/gigaspeech)                    | Audiobook, podcast, YouTube | Narrated, spontaneous | 10000       | ❌      | ✅           | apache-2.0      | Robustness over multiple domains |
+| [SPGISpeech](https://huggingface.co/datasets/kensho/spgispeech)                         | Fincancial meetings         | Oratory, spontaneous  | 5000        | ✅      | ✅           | User Agreement  | Fully formatted transcriptions   |
+| [Earnings-22](https://huggingface.co/datasets/revdotcom/earnings22)                     | Fincancial meetings         | Oratory, spontaneous  | 119         | ✅      | ✅           | CC-BY-SA-4.0    | Diversity of accents             |
+| [AMI](https://huggingface.co/datasets/edinburghcstr/ami)                                | Meetings                    | Spontaneous           | 100         | ✅      | ✅           | CC-BY-4.0       | Noisy speech conditions          |
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:432]: I just went with the ESB benchmark datasets (we've built the Google Colab script around this)

File: audio-datasets.md [hunk 24]
```diff
@@ -0,0 +1,565 @@
...
+
+The following dataset descriptions are largely taken from the [ESB Benchmark](https://arxiv.org/abs/2210.13352) paper.
+
+#### [LibriSpeech ASR](https://huggingface.co/datasets/librispeech_asr)
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:436]: IMO our first priority should be getting people to use the 'popular' ASR datasets - we see in the Whisper paper that they use HF models for their baselines, but torchaudio for their datasets. This suggests that people aren't aware that we have the top audio datasets available on the Hub -> let's try and change that by making them more visible!

File: audio-datasets.md [hunk 25]
```diff
@@ -0,0 +1,565 @@
...
+           'array': array([0.0005188 , 0.00085449, 0.00012207, ..., 0.00125122, 0.00076294,
+       0.00036621], dtype=float32), 
+           'sampling_rate': 16000}}
+```
+
+Great! We can see that we've only got the two required columns `text` and `audio`. We can also see that the audio is 
+sampled at a sampling rate of 16kHz. We can set the audio inputs to our desired sampling rate using 🤗 Datasets' 
+[`cast_column`](https://huggingface.co/docs/datasets/package_reference/main_classes.html?highlight=cast_column#datasets.DatasetDict.cast_column) 
+method. This operation does not change the audio in-place, but rather signals to `datasets` to resample audio samples 
+_on the fly_ the first time they are loaded. The following code cell will set the sampling rate to 8kHz:
```
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md]: Good point! Clarified in https://github.com/huggingface/blog/pull/636/commits/9c41865f168ca661aa45c7ac09907f0aa2944d52

> maybe worth mentioning somewhere that indexing a single element is preferred over indexing a column
Think we can omit this from writing, but strongly reinforce it through our code statements

File: audio-datasets.md (hunk 4 above)
Comment (sanchit-gandhi) [reply audio-datasets.md:446]: Resolved in https://github.com/huggingface/blog/pull/636/commits/2661666b7567417021436502a7d253b84b9284bb

File: audio-datasets.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sanchit-gandhi) [reply audio-datasets.md:41]: Paraphrased it, but would still like to make the point that Datasets is the go-to place!

https://github.com/huggingface/blog/pull/636/commits/fd950f60da53fa109eb2813c009fd147e6096d86

File: audio-datasets.md [hunk 26]
```diff
@@ -0,0 +1,580 @@
...
+#### [FLEURS](https://huggingface.co/datasets/google/fleurs)
+FLEURS (Few-shot Learning Evaluation of Universal Representations of Speech) is a dataset for evaluating speech recognition 
+systems in 102 languages, including many that are classified as 'low-resource'. The data is derived from the [FLoRes-101](https://arxiv.org/abs/2106.03193) 
+dataset, a machine translation corpus with 3001 sentence translations from English to 101 other languages. Native 
+speakers are recorded narrating the sentence transcriptions in their native languages. An \\(n\\)-way parallel corpus of 
+speech translation data is constructed by pairing the recorded audio data with the sentence transcriptions for each of the 
+101 languages. The training sets contain approximately 10 hours of supervised audio-transcription data per source-target 
+language combination.
+
+### Audio Classification
```
Comment (polinaeterna) [inline audio-datasets.md:558]: oh I've missed that we also have [Speech Commands](https://huggingface.co/datasets/speech_commands) and [MLSpokenWords](https://huggingface.co/datasets/MLCommons/ml_spoken_words) for keyword spotting

**Review by polinaeterna — APPROVED** [review]
thank you for the streaming fix! looks more correct from the technical point of view now to me :) 

I have left a few small nits and a suggestion to add SpeechCommands and MLSpokenWords to the list of audio classification datasets. the first one used to be a classic one as far as I know and the second one is multilingual and big!

File: audio-datasets.md (hunk 26 above)
Comment (sanchit-gandhi) [reply audio-datasets.md:571]: Thanks for the suggestions! Resolved in https://github.com/huggingface/blog/pull/636/commits/2dbeaf12f4ae2c2e74c637636abf6e69bada7dba

**Pull request merged**